mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -471,7 +504,10 @@ class ExperimentView extends Component {
   }
 
   onLifecycleFilterInput(newLifecycleInput) {
-    this.setState({ lifecycleFilterInput: newLifecycleInput }, this.onSearch);
+    this.setState({ lifecycleFilterInput: newLifecycleInput }, () => {
+      this.snapshotComponentState();
```

Comment:
Why do we need to do this on the result of setState?

Reply:
Good question, the TL;DR is that `this.onSearch()` on the following line actually causes the ExperimentView to be reconstructed / it doesn't seem `componentDidUpdate` gets called, so we snapshot the state here first.

More detail: `this.onSearch` calls into `ExperimentPage.onSearch`, which setsState on `ExperimentPage`, which triggers `render` on `ExperimentPage`, which renders a `RequestStateWrapper` that shows a spinning wheel until the REST API requests finish, then reconstructs and shows an `ExperimentView`. Since the DOM tree changes (from a div [here](https://github.com/mlflow/mlflow/blob/e86f1084fade59e5d75d617201af90eb2eb27283/mlflow/server/js/src/components/RequestStateWrapper.js#L67) to an ExperimentView [here](https://github.com/mlflow/mlflow/blob/e86f1084fade59e5d75d617201af90eb2eb27283/mlflow/server/js/src/components/RequestStateWrapper.js#L60)), React tears down the tree & reconstructs the ExperimentView

Reply:
Hmm, maybe if we saved on componentWill/DidUnmount? I just worry if we have to hook into these places, it's likely we'll forget one down the line.

Reply:
Yeah good point, componentWillUnmount seems like a good place for this, will make that change